api/service/version_endpoint_service.go

```diff
@@ -60,24 +75,121 @@ type endpointService struct {
 	environment          string
 	monitoringConfig     config.MonitoringConfig
 	loggerDestinationURL string
+	jobProducer          queue.Producer
 }
 
-func NewEndpointService(clusterControllers map[string]cluster.Controller,
-	imageBuilder imagebuilder.ImageBuilder,
-	storage storage.VersionEndpointStorage,
-	deploymentStorage storage.DeploymentStorage,
-	environment string,
-	monitoringConfig config.MonitoringConfig,
-	loggerDestinationURL string) EndpointsService {
+func NewEndpointService(params EndpointServiceParams) EndpointsService {
 	return &endpointService{
-		clusterControllers:   clusterControllers,
-		imageBuilder:         imageBuilder,
-		storage:              storage,
-		deploymentStorage:    deploymentStorage,
-		environment:          environment,
-		monitoringConfig:     monitoringConfig,
-		loggerDestinationURL: loggerDestinationURL,
+		clusterControllers:   params.ClusterControllers,
+		imageBuilder:         params.ImageBuilder,
+		storage:              params.Storage,
+		deploymentStorage:    params.DeploymentStorage,
+		environment:          params.Environment,
+		monitoringConfig:     params.MonitoringConfig,
+		loggerDestinationURL: params.LoggerDestinationURL,
+		jobProducer:          params.JobProducer,
+	}
+}
+
+type EndpointJobArgs struct {
+	Endpoint *models.VersionEndpoint
+	Model    *models.Model
+	Version  *models.Version
+	Project  mlp.Project
+}
+
+// ExecuteDeployment will create inferenceservice based on job that already queued
+func (k *endpointService) ExecuteDeployment(job *queue.Job) error {
```

Comment:
Similar to comment https://github.com/gojek/merlin/pull/107/files#r601994686 . I think `ExecuteDeployment` shouldn't be part of the service. It's better to split it as separate class and closely related to `queue` package.

Reply:
package queue will handle registration of job and consume the job, but the logic to consume the job itself is handled in relevant package. IMO version_endpoint_service or prediction_job_service still relevant to deploy the job, or do we need another file for deployment only let say deployment_service.go?

Reply:
Yeah, better to separate these into its own package. probably under queue/work. We can name it deployment.go and batch_prediction.go